### PR TITLE
School report cards

### PR DESCRIPTION
**Pull request by RolfBates**
This branch adds the link to school report cards. Closes issue #287. Also adds state_id field to schoolprofile table in database.

### CODE REVIEW

**Comment by jtf621** [comment]
@RolfBates I was testing this on a copy of the production database and got a KeyError when migrating.  It looks like you are not handling the case when a new school was added to the db that is not in your migration.

Running migrations:
  Rendering model states... DONE
  Applying schools.0038_schoolprofile_state_id...Traceback (most recent call last):
  File "manage.py", line 13, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/core/management/__init__.py", line 330, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/core/management/base.py", line 393, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/core/management/base.py", line 444, in execute
    output = self.handle(*args, **options)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/core/management/commands/migrate.py", line 222, in handle
    executor.migrate(targets, plan, fake=fake, fake_initial=fake_initial)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/db/migrations/executor.py", line 110, in migrate
    self.apply_migration(states[migration], migration, fake=fake, fake_initial=fake_initial)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/db/migrations/executor.py", line 148, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/db/migrations/migration.py", line 115, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/Users/jeremy/.virtualenvs/school_navigator/lib/python3.4/site-packages/django/db/migrations/operations/special.py", line 183, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/Users/jeremy/code/codefordurham/school-navigator/schools/migrations/0038_schoolprofile_state_id.py", line 78, in insert_state_ids
    school.state_id = id_map[school.school_id]
KeyError: 247

File: frontend/app/templates/details.html [hunk 1]
```diff
@@ -186,9 +186,9 @@
             </div>
         </div>
         <div class="row map-row">
-            <div class="col-md-6 coming-soon">
+            <div class="col-md-6">
                 <h2>State Report Card</h2>
-                <p><a href="{{ school.report_card_link }}">To view the state report card for this school, click here.</a></p>
+                <p><a ng-href="{{report_card_link()}}" target="_blank">To view the state report card for this school, click here.</a></p>
```
Comment (alexlemann) [inline frontend/app/templates/details.html]: Better handling of null ``state_id`` ?

**Comment by RolfBates** [comment]
@jtf621 Yeah I'll set that field to an empty string if the state id code isn't known for future additions. What schools are you adding in your migration? I can add those into the dictionary so that their state id is populated.

@alexlemann I was thinking about that. I left it out in the excitement to get it wrapped up. For that case, I could do a couple things: the whole section could be hidden, the link could be empty or link to top of the page. I'm thinking it would work best to hide the link and instead show a message that the report card is unavailable.

**Comment by jtf621** [comment]
@RolfBates, the schools we added are one issue (I will have them fixed via the admin so you can get your pull request done without worrying about syncing with changing data.

The issuse I would like to see addressed is the case where your dict thows a KeyError:

consider 

``` id_map.get(school.school_id, None)
```

vs.
``id_map[school.school_id]```

**Comment by victor-rocha** [comment]
this looks fine, I am going to be mergin this. I do agree with Jeff that school.state_id = id_map.get(school.school_id, '') would have been more idiomatic though.

**Pull request merged**